### ISSUE TITLE
avoid_returning_null_for_void false positives

### ISSUE THREAD

**Issue by davidmorgan** [dart-lang/linter#1787]
The lint fires for this:

```dart
  void clearFilter() => _filteredItems = null;
```

Effective Dart explicitly calls this out as something that's reasonable to do

https://dart.dev/guides/language/effective-dart/usage#consider-using--for-simple-members
"You can also use => on members that don’t return a value."
```
set x(num value) => center = Point(value, center.y);
```

so I think an update is in order to not fire on `=>` methods.

What do you think, please?

**Comment by davidmorgan** [comment]
Great, thanks!
